pkg/k8s/kubectl.go

```diff
@@ -15,45 +13,27 @@ import (
 
 type Kubectl interface {
 	Version() ([3]int, error)
-	StartProxy(potentialErrorWhenStartingProxy chan error, port int) error
-	UrlFor(namespace string, extraPathStartingWithSlash string) (*url.URL, error)
-	ProxyPort() int
 	healthcheck.StatusChecker
 }
 
 type kubectl struct {
-	sh        shell.Shell
-	proxyPort int
+	sh shell.Shell
 }
 
 const (
 	KubernetesDeployments               = "deployments"
 	KubernetesPods                      = "pods"
-	kubectlDefaultProxyPort             = 8001
 	kubectlDefaultTimeout               = 10 * time.Second
 	portWhenProxyNotRunning             = -1
```

Comment:
I think you can remove `kubectlDefaultTimeout` and `portWhenProxyNotRunning` too.